### PR TITLE
vlan: add VLAN plugin

### PR DESCRIPTION
**Pull request by dcbw**
Add a pretty simple plugin that just creates a VLAN interface in the given namespace off a given master.

### CODE REVIEW

**Comment by dcbw** [comment]
@containernetworking/cni-maintainers

**Comment by bboreham** [comment]
Should this go in https://github.com/containernetworking/plugins ?

**Comment by dcbw** [comment]
@bboreham I must have missed that, are we generally moving plugins over there now?

**Comment by bboreham** [comment]
It was discussed at the Community Sync meeting; we have three things that ought to release on different schedules: the spec, the library and the plugins.  I don't think anyone's fired up to move the existing plugins, but it seems like a good idea to put new ones there.

Maybe we could just discuss whether we think this particular plugin should be forever maintained by the CNI maintainers before you go ahead and open that PR, however.

File: plugins/main/vlan/vlan.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	}
+
+	if err := netlink.LinkAdd(v); err != nil {
+		return nil, fmt.Errorf("failed to create vlan: %v", err)
+	}
+
+	err = netns.Do(func(_ ns.NetNS) error {
+		err := ip.RenameLink(tmpName, ifName)
+		if err != nil {
+			return fmt.Errorf("failed to rename vlan to %q: %v", ifName, err)
```
Comment (squeed) [inline plugins/main/vlan/vlan.go]: Probably worth trying to clean up the failed link here too.

**Comment by dcbw** [comment]
Closed in favor of https://github.com/containernetworking/plugins/pull/2

**Pull request closed without merging**